pilot/pkg/config/kube/gateway/supported_features.go

```diff
@@ -0,0 +1,34 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gateway
+
+import (
+	k8sv1 "sigs.k8s.io/gateway-api/apis/v1"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+
+	"istio.io/istio/pkg/util/sets"
+)
+
+var SupportedFeatures = suite.AllFeatures
```

Comment:
I think this might be your problem. `suite.AllFeatures` contains elements that aren't part of the kubebuilder validation list (see [here](https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1/gatewayclass_types.go#L269)).
I'm currently facing this problem as I try to implement `SupportedFeatures` at envoy.

I haven't found a list of defined supported features in gateway-api's that can filter out things from `suite.AllFeatures` that aren't in the validation list. I think it can be helpful to add such a list.

Reply:
So far it seems to be it, thanks @levikobi !